java/client/src/main/java/glide/api/models/commands/stream/StreamGroupOptions.java

```diff
@@ -0,0 +1,65 @@
+/** Copyright GLIDE-for-Redis Project Contributors - SPDX Identifier: Apache-2.0 */
+package glide.api.models.commands.stream;
+
+import glide.api.commands.StreamBaseCommands;
+import java.util.ArrayList;
+import java.util.List;
+
+/**
+ * Optional arguments for {@link StreamBaseCommands#xgroupCreate(String, String, String,
+ * StreamGroupOptions)}
+ *
+ * @see <a href="https://valkey.io/commands/xgroup-create/">valkey.io</a>
+ */
+public final class StreamGroupOptions {
+
+    public static final String MAKE_STREAM_REDIS_API = "MKSTREAM";
+    public static final String ENTRIES_READ_REDIS_API = "ENTRIESREAD";
+
+    /** If the stream doesn't exist, creates a new stream with a length of 0. */
+    boolean makeStream;
+
+    /**
+     * An arbitrary ID (that isn't the first ID, last ID, or the zero <code>"0-0"</code>. Use it to
+     * find out how many entries are between the arbitrary ID (excluding it) and the stream's last
+     * entry.
+     *
+     * @since Redis 7.0.0
+     */
+    String entriesRead;
+
+    public StreamGroupOptions(Boolean makeStream) {
+        this.makeStream = makeStream;
+    }
+
+    public StreamGroupOptions(String entriesRead) {
+        this.makeStream = false;
+        this.entriesRead = entriesRead;
+    }
+
+    public StreamGroupOptions(Boolean makeStream, String entriesRead) {
+        this.makeStream = makeStream;
+        this.entriesRead = entriesRead;
+    }
```

Comment:
Please add descriptions to the constructors. These do not require complete docs.
You can move/copy the from the member declarations.